api/v1alpha1/openapi.yaml

```diff
@@ -2426,6 +2470,12 @@ paths:
                 $ref: '#/components/schemas/Error'
 components:
   schemas:
+    DeviceDesiredDecomState:
+      type: string
+      enum:
+        - "DefaultDecom"
+        - "CustomDecom"
```

Comment:
We know we need custom actions anyway, so depending on the incremental effort (e.g. reusing hooks or not), why not.

To me the question is more the modelling: Isn't the parameter more about "how far" to decommission, i.e. to simply unenroll or to factory reset or to wipe/burn? Whether that's done by the agent or custom scripts or both is a different question and I'm not sure it belongs here.

Reply:
Ok, I think I'm confused about the use of this field now:
If this is a user-defined action, we cannot know whether/how that action performs reset, wipe, ...).
Also, we can define a well-known path in the device's filesystem (e.g. `/usr/flightctl/hooks.d/decommission.sh`) that - if present - gets executed at a certain point in the decom process. Means that'd replace "CustomDecom".
Therefore, I would have assumed this field to tell whether the agent shall just unenroll or go further?

Reply:
I was thinking for this first iteration, we only have one setting that enables our only / default way of decommissioning: unenroll the device and send a signal to the device to do hardcoded / default actions, which means wipe the cert and key.

Later as we want to support more specific and custom options, we could add other settings: 'unenrollOnly', 'runCustomOnly', 'runDefaultAndCustom', etc., or whatever it is that we end up wanting to support. But not yet having to decide exactly what those distinctions are means we can support a set of initial default actions more quickly.

Reply:
We could decide to name `default` something more descriptive that fits what we are going to support initially.

Reply:
@fzdarsky I changed this to `DefaultUnenroll` to be more descriptive - thoughts?
In the future we could add `DefaultWipe`, `CustomWipe`, or whatever makes sense.

Reply:
Naming is important in API design; names should be self-explanatory and naming something "default" isn't. And it seems we're not sure what that default is and which direction we want to take for the other values yet, so this seems merely a placeholder.

How about we first try to figure out what `decommissioningType` is? "Type" again is very unspecific. Are we thinking of
* a decommissioningTarget, e.g. Unenroll --> Unenroll+Wipe --> Unenroll+Wipe+FactoryReset --> Unenroll+Wipe+FactoryReset+Block
* a decommissioningPolicy, e.g. Unenroll plus whether or not to a) wipe data, b) run custom scripts, c) ...
* an Action? a Strategy? a Sequence?
Does it even make sense to give an operator the choice not to run a built-in or user-defined action? Arguably, if there is a user-defined action, it's been placed there deliberately by some security engineer and is mandatory.

My hunch is we're talking about a Target. And then your suggestion "UnenrollOnly" would be a valid target and maybe even a good default. It may also be the only target that's completely built-in, with others having at least one user-defined component.

Reply:
These are good questions to refine the idea, thanks.

>Does it even make sense to give an operator the choice not to run a built-in or user-defined action?

I'm not sure it does. And if each device is pre-loaded with the actions it should take upon decommissioning, then maybe there are no decisions to be made at this step...